### PR TITLE
Issue 38

### PR DESCRIPTION
**Pull request by ubaldop**
Hi @camilosampedro, 

this PR solves your issue #38 about the need to create a unit test for `LaboratoryController`.
Please, verify if my commit is enough.

Moreover, within this PR I got the chance to [remove a couple of compilation warnings](https://github.com/camilosampedro/Aton/commit/b7c409787d64aacb20495d4484482d8944f5e503) occurring on my machine.

Hope it helps, cheers!

Have a nice weekend,

@P3trur0

### CODE REVIEW

File: app/model/table/ComputerTable.scala [hunk 1]
```diff
@@ -16,7 +16,7 @@ class ComputerTable(tag: Tag) extends Table[Computer](tag, "computer") {
   def name = column[Option[String]]("name")
 
   // Room foreign key
-  def room: ForeignKeyQuery[RoomTable, Room] = foreignKey("room_id", roomId, TableQuery[RoomTable])(_.id, onUpdate = ForeignKeyAction.Restrict, onDelete = ForeignKeyAction.Cascade)
+  def room: ForeignKeyQuery[RoomTable, Room] = foreignKey("room_id", roomId, TableQuery[RoomTable])(_.id?, onUpdate = ForeignKeyAction.Restrict, onDelete = ForeignKeyAction.Cascade)
```
Comment (camilosampedro) [inline app/model/table/ComputerTable.scala:19]: `?` is a good contribution, I read about optional column checking with `?` but I still don't figure how it does the check. 😕

**Review by camilosampedro — APPROVED** [review]
You made a great job, that's a very good unit test. I have a question for you: What do you think about having methods that create unit tests (Not creating them manually)? Are they hard to understand? Do they follow simple good practices? They are basically the first _"intentionally good"_ unit tests that I create.

**Comment by camilosampedro** [comment]
The checks are failing, I will be checking them on this moment. 😃

**Comment by ubaldop** [comment]
@camilosampedro looking at the TravisCI stacktrace it seems there is some problem with the MySQL authentication.

Did you consider to setup a Docker environment before your CI starts?
In this case Travis should start a container for MySQL where it is possible to check all your tests.

My assumption here is that since Travis does not have a MySQL setup, it has no chance to execute the unit tests on the database.

---

**About your question on tests, instead** my personal opinion is that tests should be as more understandable and replicable as possible.

If you make them to much complex you have to deal also with test complexities that, instead, should be reduced at minimum to allow developers concentrate on the actual meaning of the business logic.

Is fine to have utility methods to create mocks, maybe, but I'd leave the actual testing logic in small methods.

AFAIK, the Unit Testing movement asserts that "_A unit test focuses on a single “unit of code” – usually a function in an object or module. By making the test specific to a single function, the test should be simple, quick to write, and quick to run_".

This [blog post (just Googled it, btw)](https://codeutopia.net/blog/2015/03/01/unit-testing-tdd-and-bdd/) seems an interesting starting point!

File: app/model/table/ComputerTable.scala (hunk 1 above)
Comment (ubaldop) [reply app/model/table/ComputerTable.scala:19]: It is a lifting method offered by Slick.

You can see the API definition [here](http://slick.lightbend.com/doc/3.1.1/api/index.html#slick.lifted.BaseColumnExtensionMethods@?:slick.lifted.Rep[Option[P1]])

In this case the `Rep[Long]` instance is lifted to `Rep[Option[Long]]`.

Hope it helps! :smile:

**Comment by camilosampedro** [comment]
Thank you @P3trur0 ! I will be working on this, I had a very busy weekend 😅
